packages/fx-core/src/component/utils/backupFiles.ts

```diff
@@ -133,6 +133,36 @@ export async function backupFiles(
   return ok(undefined);
 }
 
+// append folder path to .gitignore under the project root.
+export async function addPathToGitignore(
+  projectPath: string,
+  ignoredPath: string,
+  log: LogProvider
+): Promise<void> {
+  const relativePath = path.relative(projectPath, ignoredPath).replace(/\\/g, "/");
+  await addItemToGitignore(projectPath, relativePath, log);
+}
+
+// append item to .gitignore under the project root.
+async function addItemToGitignore(
+  projectPath: string,
+  item: string,
+  log: LogProvider
+): Promise<void> {
+  const gitignorePath = path.join(projectPath, ".gitignore");
+  try {
+    await fs.ensureFile(gitignorePath);
+
+    const gitignoreContent = await fs.readFile(gitignorePath, "UTF-8");
+    if (gitignoreContent.indexOf(item) === -1) {
+      const appendedContent = os.EOL + item;
+      await fs.appendFile(gitignorePath, appendedContent);
+    }
+  } catch {
+    log.warning(`[core] Failed to add '${item}' to '${gitignorePath}', please do it manually.`);
```

Comment:
I remember there was a PR that removes the `[core]` prefix in logs. Can you double confirm whether we still need this prefix?

Reply:
This code is copied from v2 migration file. 
The backupFiles.ts refers to it which should be removed as well with provisionUtil.
To remove code by component, we just copy the code for removing the dependency.